### PR TITLE
[Feature] Add ViT backbone for action recognition.

### PR DESCRIPTION
**Pull request by SuX97**
https://github.com/lucidrains/vit-pytorch.

### CODE REVIEW

File: mmaction/models/backbones/vit.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                 num_classes,
+                 dim,
+                 depth,
+                 heads,
+                 mlp_dim,
+                 channels=3,
+                 dropout=0.,
+                 emb_dropout=0.):
+        super().__init__()
+        assert image_size % patch_size == 0, 'Image dimensions must be divisible by the patch size.'  # noqa: E501
```
Comment (innerlee) [inline mmaction/models/backbones/vit.py:117]: This can be multiple lines

File: mmaction/models/backbones/vit.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+        self.heads = heads
+        self.scale = dim**-0.5
+
+        self.to_qkv = nn.Linear(dim, dim * 3, bias=False)
+        self.to_out = nn.Sequential(nn.Linear(dim, dim), nn.Dropout(dropout))
+
+    def forward(self, x, mask=None):
+        b, n, _, h = *x.shape, self.heads  # noqa: F841
+        qkv = self.to_qkv(x).chunk(3, dim=-1)
+        q, k, v = map(lambda t: rearrange(t, 'b n (h d) -> b h n d', h=h), qkv)
```
Comment (innerlee) [inline mmaction/models/backbones/vit.py:57]: can this be implemented by pytorch's einsum?

Comment (SuX97) [reply mmaction/models/backbones/vit.py:57]: Not really, `rearrange` can be replaced by `view` or `reshape`, it's not for dot-product or summing, but a more readable tensor manipulation. Will fix if the ckpt is ready.

File: mmaction/models/heads/null_head.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import torch.nn as nn
+
+from ..registry import HEADS
+
+
+@HEADS.register_module()
+class NullHead(nn.Module):
```
Comment (dreamerlin) [inline mmaction/models/heads/null_head.py:7]: `class Identity` may be better

**Comment by SuX97** [comment]
Closed. It is hard to converge.

**Pull request closed without merging**